src/hooks/useAppStore.ts

```diff
@@ -0,0 +1,19 @@
+import { create } from "zustand";
```

Comment:
I feel that zustand can be used for multiple store instances. You basically can create different hooks and use them to bundle functionality and then share it with different modules. 
If that's the case, maybe we could rename this to be more stations specific rather than AppStore. If not, ignore my comment.

Reply:
From what I've found so far, it seems that until a couple years ago the suggestion was to use always and only one single store as the single source of truth for the state of the whole app. Zustand is an unopinionated tool, there are no strict rules, and although its implementation is not blocking having many stores, it is still suggested to keep one main store (zustand's official doc about recommented patterns https://github.com/pmndrs/zustand/blob/main/docs/guides/flux-inspired-practice.md ). When reading through documents and articles (trying not to be too biased), having many stores is not necessary a bad design, especially when the app grows and the store becomes huge. It could help logically group the components' state where needed. However, I'm still not convinced that having many stores is a good design for a few reasons. I've listed some of my thoughts below:
- although the stores can be isolated, there could be overrides/collisions causing issues, there is no way to practically safeguard this
- having multiple stores, doesn't mean necessarily different exposed hooks, they will need to be sliced and then combined back to one main store (at least using zustand)
- it seems the only real benefit would be improving the performance, but as mentioned this comes with slicing the main store, which comes with extra work and effort.

To summarise, personally I'd prefer to keep one store because logically the whole concent is around the station, considering also that our application is very light and small as of now. If we still want to explore the other approach I'm happy to give it a go.

Reply:
Wow, thank you for the detailed answer. I do agree with going forward with one store since the app won't be so big!